packages/gestalt/src/Badge.tsx

```diff
@@ -71,52 +82,123 @@ export default function Badge({ position = 'middle', text, type = 'info', toolti
     'warning': 'workflow-status-warning',
     'success': 'check-circle',
     'neutral': 'lock',
-    'recommendation': 'performance-plus',
+    'recommendation': 'sparkle',
     'darkWash': 'info-circle',
     'lightWash': 'info-circle',
   });
 
+  const COLOR_ICON_MAP = Object.freeze({
+    'info': 'info',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
+  const COLOR_TEXT_MAP = Object.freeze({
+    'info': 'shopping',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
   let styleType: TypeOptions | InteractiveTypeOptions = type;
 
   if (shouldUseTooltip) {
     styleType = `interactive-${type}`;
   }
 
-  const csBadge = cx(styles.Badge, styles[position], styles[styleType]);
+  const { handleOnBlur, handleOnFocus, handleOnMouseEnter, handleOnMouseLeave, isFocused } =
+    useInteractiveStates();
+
+  const cxStyles = useMemo(
```

Comment:
Is there any benefit for `useMemo` here, I believe `cx` just outputs a string and it may not be a heavy lift operation

Reply:
It seems like it runs on almost every property change as well

Reply:
DONE